src/BioformatsLoader.jl

```diff
@@ -233,20 +233,49 @@ end
 
 get_bf_path() = joinpath(dirname(@__DIR__), "deps", "bioformats_package.jar")
 
+function set_memory(memory=1024)
+    if memory > 0
+        # There should be a distinct memory option in JavaCall
+        # Find other memory options and delete them
+        for opt in JavaCall.opts
+            if startswith(opt, "-Xmx")
+                delete!(JavaCall.opts, opt)
+            end
+        end
```

Comment:
```suggestion
        filter!(opt -> startswith(opt, "-Xmx"), JavaCall.opts)
```

Reply:
This is actually does the opposite the code it replaces. The line should the negation of the above.
```julia
filter!(opt -> !startswith(opt, "-Xmx"), JavaCall.opts)
```

The `filter!` code is as such looks similar to my original for loop. The difference is that my version avoids the double negation. I think the compiler sorts it out in the end.
https://github.com/JuliaCollections/OrderedCollections.jl/blob/adb9415570faeba7e8c6d800cd743de06104e97e/src/ordered_set.jl#L58-L65

Do you still prefer the `filter!` one liner?

Reply:
Ah yes, sorry, the predicate should of coirse be negated. Yes, I know it's implemented the same way but I personally prefer wrapping for loops in functions since functions better describe intent (which you often need to explain with comments otherwise).

If the line looks too long you can write it using do notation. It's not the one-liner I'm after it's expression of intent.

Reply:
I went with `filter!(!startswith("-Xmx"), JavaCall.opts)`

Reply:
That is the way I'd prefer it, unfortunately, it's not available in 1.0 I think. Could you verify which versions that restrict us to?

Reply:
The single argument `startswith` is Julia 1.5. I changed it to be 1.0 compatible.